### PR TITLE
move quantization outside of draco

### PR DESCRIPTION
**Pull request by svensht2**
Have the application to be responsible for the quantization
Also, add support for I Frame indexing to better support frame drops

### CODE REVIEW

File: src/draco/compression/mesh/mesh_edgebreaker_encoder_impl.cc [hunk 1]
```diff
@@ -299,11 +299,11 @@ bool MeshEdgeBreakerEncoderImpl<TraversalEncoder>::EncodeConnectivity() {
   // together, unless the option |use_single_connectivity_| is set in which case
   // we break the mesh along attribute seams and use the same connectivity for
   // all attributes.
-  PSY_DRACO_PROFILE_SECTION("CreateCornerTable");
   if (use_single_connectivity_) {
     PSY_DRACO_PROFILE_SECTION("CreateCornerTableFromAllAttributes");
     corner_table_ = CreateCornerTableFromAllAttributes(mesh_);
   } else {
+	PSY_DRACO_PROFILE_SECTION("CreateCornerTable");
```
Comment (zy53012127) [inline src/draco/compression/mesh/mesh_edgebreaker_encoder_impl.cc]: indention.

File: src/draco/psy/psy_draco.h [hunk 2]
```diff
@@ -65,7 +65,9 @@ struct PSY_DRACO_API Header
 {
     uint8_t mMajorVersion;
     uint8_t mMinorVersion;
+	float mDecodeMultiplier;
     MeshType mMeshType;
+	uint32_t mIFrameIndex;
```
Comment (zy53012127) [inline src/draco/psy/psy_draco.h]: indention.

File: src/draco/psy/psy_draco_decoder.cpp [hunk 3]
```diff
@@ -228,14 +249,18 @@ class MeshDecompression::Impl
...
     std::unique_ptr<MeshEdgeBreakerDecompression> mpMeshDecompression;
     ::draco::Status mStatus;
 };
 
 MeshDecompression::MeshDecompression(const MeshDecompression&) : mpImpl(nullptr) {};
-MeshDecompression& MeshDecompression::operator=(const MeshDecompression&) { return *this; };
+MeshDecompression& MeshDecompression::operator=(const MeshDecompression&)
+{
+    return *this;
+};
```
Comment (zy53012127) [inline src/draco/psy/psy_draco_decoder.cpp]: Do you know why we have this weird copy constructor?

**Review by zy53012127 — COMMENTED** [review]
some weird indention in general.

File: src/draco/psy/psy_draco_decoder.cpp (hunk 3 above)
Comment (svensht2) [reply src/draco/psy/psy_draco_decoder.cpp]: you mean assignment operator, and I do not @nbcong any input why you added it?

**Comment by svensht2** [comment]
@zy53012127 fixed ident issues and removed weird copy and assignment operators

File: src/draco/psy/psy_draco_decoder.cpp [hunk 4]
```diff
@@ -228,15 +249,13 @@ class MeshDecompression::Impl
...
+    uint32_t mCurrIFrameIndex;
     std::shared_ptr<::draco::Mesh> mpMesh;
     std::shared_ptr<::draco::DecoderBuffer> mpBuffer;
     std::unique_ptr<MeshEdgeBreakerDecompression> mpMeshDecompression;
     ::draco::Status mStatus;
 };
 
-MeshDecompression::MeshDecompression(const MeshDecompression&) : mpImpl(nullptr) {};
-MeshDecompression& MeshDecompression::operator=(const MeshDecompression&) { return *this; };
-
```
Comment (zy53012127) [inline src/draco/psy/psy_draco_decoder.cpp:239]: It feels like @nbcong wants to disable the copy? These two functions are essentially doing nothing.
I guess you could make `MeshDecompression::MeshDecompression(const MeshDecompression&) = delete` to see if anything fails compiling.

Comment (svensht2) [reply src/draco/psy/psy_draco_decoder.cpp:239]: I think putting them in private without an implementation will do the trick, I tested that everything compiles with this code and works

Comment (zy53012127) [reply src/draco/psy/psy_draco_decoder.cpp:239]: Ahh I see they're private already. I thought this was the header file. 👍

**Comment by svensht2** [comment]
I'll wait for @nbcong approval

File: src/draco/psy/psy_draco_decoder.cpp (hunk 4 above)
Comment (nbcong) [reply src/draco/psy/psy_draco_decoder.cpp:239]: Sorry, I still not understand what the problems if we make them private and have implementations in .cpp file?
@zy53012127 yes, I want to disable the copy!

**Comment by nbcong** [comment]
@svensht2 I actually get confuses from this PR. The original purpose, which I would like to have, is making the wrapper more general usage. In case we want to have a version same as your changing, I think we can create another api or another class inherits from the current codes.

File: src/draco/psy/psy_draco.h [hunk 5]
```diff
@@ -65,7 +65,9 @@ struct PSY_DRACO_API Header
 {
     uint8_t mMajorVersion;
     uint8_t mMinorVersion;
+    float mDecodeMultiplier;
     MeshType mMeshType;
+    uint32_t mIFrameIndex;
```
Comment (nbcong) [inline src/draco/psy/psy_draco.h:71]: can you update change logs too?

File: src/draco/psy/psy_draco_decoder.cpp (hunk 4 above)
Comment (svensht2) [reply src/draco/psy/psy_draco_decoder.cpp:239]: we don't need an implementation to just disable copy

File: src/draco/psy/psy_draco.h (hunk 5 above)
Comment (svensht2) [reply src/draco/psy/psy_draco.h:71]: will do

**Comment by svensht2** [comment]
@nbcong my main change was to change the vertex from being a floating point to being an already quantized integer. Are you suggesting we support both this and the original floating point vertex + quantization parameters? I'm worried that the other code path will not be properly supported and tested since we aren't using it

**Comment by nbcong** [comment]
@svensht2 I see that, and so I suggested to have another api to support both floating point and integer vertex position. However if you think we only need one, please go ahead; I don't have objection now.

**Comment by svensht2** [comment]
I don't see us needing to support both for the foreseeable future, we can add the support back if that ever changes

**Comment by svensht2** [comment]
updated change log

**Comment by nbcong** [comment]
👍

**Pull request merged**